src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -20,6 +20,8 @@
 #include <sys/types.h>
 #include <sys/event.h>
 #include <sys/time.h>
+#else
+#include <sys/poll.h>
```

Comment:
Is the sys/poll.h guaranteed to exist on systems that don't have epoll or kqueue? If not, then I would also recommend adding a detection for the header presence to configure.cmake and include the header only if it exists.